mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -123,30 +130,60 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 	// provide this basic functionality before.
 	b.Lock()
 	defer b.Unlock()
-	controller, found := b.controllers[path]
+	_, found := b.controllers[path]
 	if !found {
 		clientset, err := b.newClientFn(path, env)
 		if err != nil {
 			return nil, fmt.Errorf("could not build kubernetes client: %v", err)
 		}
-		controller = newCacheController(clientset, refresh, env)
-		env.ScheduleDaemon(func() { controller.Run(stopChan) })
-		// ensure that any request is only handled after
-		// a sync has occurred
-		env.Logger().Infof("Waiting for kubernetes cache sync...")
-		if success := cache.WaitForCacheSync(stopChan, controller.HasSynced); !success {
-			stopChan <- struct{}{}
-			return nil, errors.New("cache sync failure")
+		controller, err = getNewCacheController(b, clientset, env)
```

Comment:
I don't understand your comment. The new function "getNewCacheController" just puts the deleted lines 132-139 into a function so that it can be called here (creating the primary cluster's controller) and if a secondary cluster (in a multi-cluster environment) is created.

Reply:
I just do not like this name `getNewCacheController`, how about call it `NewCacheController`?

Reply:
NewCacheController would indicate an external function to this package, which it is not. newCacheController is already defined in cache.go. I used createCacheController as the callback function name when a remote cluster is found. Ran out of obvious verb-noun combinations. getNewCacheController() is getting the newly created controller from newCacheController().

Reply:
would `startNewController` or `runNewController` perhaps be clearer and address this concern?

Reply:
runNewController it is.